scripts/west_commands/run_common.py

```diff
@@ -165,21 +198,116 @@ def do_run_common(command, user_args, user_runner_args, domains=None):
             # Get the user specified domains.
             domains = load_domains(build_dir).get_domains(user_args.domain)
 
-    if len(domains) > 1 and len(user_runner_args) > 0:
-        log.wrn("Specifying runner options for multiple domains is experimental.\n"
-                "If problems are experienced, please specify a single domain "
-                "using '--domain <domain>'")
+    if len(domains) > 1:
+        if len(user_runner_args) > 0:
+            log.wrn("Specifying runner options for multiple domains is experimental.\n"
+                    "If problems are experienced, please specify a single domain "
+                    "using '--domain <domain>'")
+
+        # Process all domains to load board names and populate flash runner
+        # parameters.
+        board_names = []
+        for d in domains:
+            if d.build_dir is None:
+                build_dir = get_build_dir(user_args)
+            else:
+                build_dir = d.build_dir
+
+            cache = load_cmake_cache(build_dir, user_args)
+            board = cache['CACHED_BOARD']
+
+            if board not in board_names:
+                board_names.append(board)
+
+            if board in board_image_count:
+                board_image_count[board].total = board_image_count[board].total + 1
+            else:
+                # This sets up the default entry for a board, no flashed images
+                # and one image in total.
+                board_image_count[board] = ImagesFlashed(0, 1)
+
+            # Load board flash runner configuration (if it exists) and store
+            # single-use commands in a dictionary so that they get executed
+            # once per unique board name.
+            if cache['BOARD_DIR'] not in processed_boards and 'SOC_FULL_DIR' in cache:
+                soc_yaml_file = Path(cache['SOC_FULL_DIR']) / 'soc.yml'
+                board_yaml_file = Path(cache['BOARD_DIR']) / 'board.yml'
+
+                # Search for flash runner configuration, board takes priority over SoC
+                try:
+                    with open(board_yaml_file, 'r') as f:
+                        data_yaml = yaml.safe_load(f.read())
+
+                except FileNotFoundError:
+                    continue
+
+                if 'run_once' not in data_yaml and 'deferred_reset' not in data_yaml:
+                    # Check SoC file
+                    try:
+                        with open(soc_yaml_file, 'r') as f:
+                            data_yaml = yaml.safe_load(f.read())
+
+                    except FileNotFoundError:
+                        continue
```

Comment:
something to handle ?

Reply:
No, hwmv1 so can be ignored

Reply:
then please add a comment in code, for example:
```suggestion
                        # Legacy boards don't have a board.yml or soc.yml.
                        continue
```